addon/-private/notifier.ts

```diff
@@ -0,0 +1,30 @@
+// this statefull class holds and notifies
+import { INotifier } from '../types/evented';
+
+export default class Notifier implements INotifier {
```

Comment:
This is a generic version of Notifier that check that callbacks admit the right arguments.

http://www.typescriptlang.org/play/?ssl=26&ssc=22&pln=26&pc=27#code/C4TwDgpgBAwghgGwQIzgYwNYB4AqUIAewEAdgCYDOUcJIA2gLoB8UAvFABQB0PcATgHMKALig4AlGxYA3APYBLMgG4AUCsJhZfYFDIQAZnACuCHWgRwKVAHKzg8-fIh9c+IqUrVajJgG8VUIFQCPIUxCTOIpzcvIJRElJQcorijGpBUGiyJGF8RmjAWhyS-hkZwAAWoVwhYaSRbFCMqhkAvulBcGRkADKh4c4caIgo6BiiMVz8QqIJrDIKZCUBZYGV1bUDfBRcYEYUFUMjqJjiLatQfBDARnwknJLzUOs7VwC2stIQfXURfEdIE4YM4rQLtUGXCAfL4-LYA0aYCY8KZxWaPBYpKClC4vGr9erbVQQjL6LScBDXKDyRoABiUVKgWGeVR2mwJrNIAkq9PkAGpecsLmUHJxcWy-hQ6PIGGxWOxhoCxoKhTiWXjfpEuBQwCE0BAOPIADRQACMIJVFyuNzu5wt7SF9qC4PKfHkAgEg2R03iytWYvxEq1uv1NPEXFJfAAouhDgqERhEnGgZNveJzU6VK0gA

Reply:
In general, types like `Function`, `String`, `Number`, etc. should not be used.